src/interface/sparse_pack.hpp

```diff
@@ -276,6 +338,10 @@ class SparsePack : public SparsePackBase {
                    const int i) const {
     return pack_(0, b, idx)(k, j, i);
   }
+  KOKKOS_INLINE_FUNCTION
+  Real &operator()(int idx, const int k, const int j, const int i) const {
+    return pack_(0, 0, idx)(k, j, i);
+  }
 
   KOKKOS_INLINE_FUNCTION
   Real &operator()(const int b, PackIdx idx, const int k, const int j,
```

Comment:
And would `PARTHENON_DEBUG_REQUIRE(!flat_, ...);` be valid here?

Reply:
yes good idea